src/parse.c

```diff
@@ -3257,6 +3295,12 @@ process_missing_ids(NetplanParser* npp, __unused GError** error)
             netdef->vlan_link = netplan_netdef_new(npp, scalar(missing->node), NETPLAN_DEF_TYPE_NM_PLACEHOLDER_, NETPLAN_BACKEND_NM);
             g_hash_table_iter_remove(&iter);
         }
+
+        /* VETH case: NetworkManager doesn't enforce the existence of the veth peer */
```

Comment:
suggestion: We might want to extend that comment a bit, to describe in more detail the relaxation of validation for this `NETPLAN_DEF_TYPE_NM_PLACEHOLDER_` special case. Future-us would be happy about that context :)